Investigate periodic flakiness with MacOS tests

Every once in a while, the MacOS tests will fail with:

```
Running tests for darwin...
LIB=llvm-project/build/lib/darwin/libclang_rt.rtsan_osx_dynamic.dylib bash ./test_common.sh
error: write on a pipe with no reader
Missing required symbol: rtsan_realtime_enter
make: *** [test] Error 1
```

See https://github.com/realtime-sanitizer/rtsan-libs/actions/runs/14560541804/job/40843240829

This library is not missing the symbol if you inspect it manually, so something else is going on. Needs more investigation

--- a/test_common.sh
+++ b/test_common.sh
@@ -7,7 +7,9 @@ if [ ! -f "$LIB" ]; then
   exit 1
 fi
 
-if ! nm -a "$LIB" | grep -q "rtsan_realtime_enter"; then
+SYMBOLS=$(nm -a "$LIB") || exit 1
+
+if ! grep -q "rtsan_realtime_enter" <<< "$SYMBOLS"; then
   echo "Missing required symbol: rtsan_realtime_enter"
   exit 1
 fi

